example_configs/wildfly-10.yaml

```diff
@@ -6,7 +6,8 @@ whitelistObjectNames:
  # see https://github.com/prometheus/jmx_exporter/issues/246#issuecomment-367573931
  # Each object in the rules below has to be added to whitelistObjectNames too !
  # note that rules use regex (like "foo.*", whereas the whitelist use globbing expressions (like "foo*")
- - "jboss.as:subsystem=messaging-activemq,server=*"
+ - "jboss.as:subsystem=messaging-activemq,server=*,jms-queue=*"
```

Comment:
`messaging-activemq,server=*,*` is what the docs indicate

Reply:
Then I have a suggestion to replace also 2 strings "jboss.as:subsystem=datasources,data-source=*,statistics=*" and "jboss.as:subsystem=datasources,xa-data-source=*,statistics=*" with "jboss.as:subsystem=datasources,statistics=*,*" (it will also find all required datasources).

What do you think of this change? (If you approve I will commit changes an add them to this pull request)

Reply:
Assuming that doesn't pull in something else we don't want, that sounds good

Reply:
I've checked, `jboss.as:subsystem=messaging-activemq,server=*,*` gets some more objects, not only the `jms-queue` and `jms-topic`
So it seems to me logical either (1) to keep both jms and datasources with `*` (your suggestion and my later suggestion). This will result in less amount of jmx calls (i.e. fewer `beanConn.queryMBeans` calls). but more obsolete objects will be received and thus more work with the pattern filters will occur. Or (2) to keep both jms and datasourses with more explicit  whitelist (which I suggested in the beginning). This will result in more jmx calls but since less MBeans will be received, less work will be done in filtering.

What seems to be more sufficient? In other words what is more "expensive": jmx call or (in my case) about 10 additional objects to filter? (being not an expert in this area I really don't know, I suspect they have pretty a similar "time price", hope you know more)

Reply:
Can you time both in your setup and see if there's a difference? If not, go with the simpler one.

Reply:
Sorry, haven't perform additional checks... So let's have the simpler variant.

Reply:
I will perform checks on WidlFly-15 (and some JBoss version) and then perform a merge here

Reply:
What was the outcome?

Reply:
Found out that monitoring on our docker image was turned off...
Sorry, once I have any news I will write it.
P.S. I've also added a pattern for java.lang.* MBeans

Reply:
Ok, this works for WildFly 15. Finally determined it (:
As for java.lang — in the white list there's a commented out line
` # - "java.lang:*"`
So I decided to add a commented line to the patterns
` # - pattern: "java.lang.*"`

Reply:
Can you remove the java.lang references? You also need to add a DCO

Reply:
Ok, I'm removing pattern line which I added (if you want I may remove both java lines, what would you prefer?).

Sorry for having to spend so much time with me... This is the first time I face the DCO. After googling I found that by this I state that changed are made by me and I have right to commit them (something like this).
Does need to add DCO means adding a comment to the git commit a line alike:
Signed-off-by: Joe Smith <joe.smith@email.com>

Reply:
Checked one of old commits, found similar line (Signed-off-by ...) as a DCO. I will add it on the last commit after a decision to remove both java.lang mentions or only one of them (which was added my be, the `pattern` one).

Reply:
After some investigations have found some more interesting details to commit. For example http-listener helps to track http communication, https-listener also helps to check https traffic. In my case this was a really important thing.